dipy/io/stateful_tractogram.py

```diff
@@ -76,18 +75,20 @@ def __init__(self, streamlines, reference, space,
         if data_per_streamline is None:
             data_per_streamline = {}
 
+        if isinstance(streamlines, Streamlines):
+            streamlines = streamlines.copy()
```

Comment:
Just to make sure we agree that this will be duplicating all the streamlines. Meaning when you create a `StateFullTractogram` from a `Streamlines` object, you might get twice the memory consumption!

Reply:
Yes, I don't want anything outside modifying the streamlines. Even if it is very rare, the goal of the StatefulTractogram is to be super strict and robust. So to me this necessary to guarantee that the state is always valid and can only be change by the object.

For example, if somebody load a Tractogram using nibabel, create a StatefulTractogram and later call to_world() or save the Tractogram using nibabel the StatefulTractogram will be corrupted.

If someone has file with 10M streamlines, do multiprocessing and loads multiple big tractograms at once, and has 8GB of RAM, well they should use Nibabel and be careful.
I am personally on the side of safety and keeping the state valid over RAM limitation (even if it is a rare occurance)